hack/generator/pkg/codegen/embeddedresources/remover.go

```diff
@@ -273,11 +273,21 @@ func findSubResourcePropertiesTypeNames(types astmodel.Types) (map[astmodel.Type
 			errs = append(errs, errors.Wrapf(err, "couldn't extract spec/status properties from %q", def.Name()))
 			continue
 		}
+
 		if specPropertiesTypeName != nil {
-			result[owner] = result[owner].Add(*specPropertiesTypeName)
+			if result[owner] == nil {
+				result[owner] = astmodel.NewTypeNameSet(*specPropertiesTypeName)
+			} else {
+				result[owner].Add(*specPropertiesTypeName)
+			}
 		}
+
 		if statusPropertiesTypeName != nil {
-			result[owner] = result[owner].Add(*statusPropertiesTypeName)
+			if result[owner] == nil {
+				result[owner] = astmodel.NewTypeNameSet(*statusPropertiesTypeName)
```

Comment:
to make these (and the one below) simpler we could have a version of `map[astmodel.TypeName]astmodel.TypeNameSet` that auto-vivifies